File: Scala/3096.scala
import scala.math.{log10, Pi, floor}

object Main {
    def main(args: Array[String]) {
        val n = scala.io.StdIn.readInt()
        
        val r = floor(n * log10(n / math.E) + log10(2 * Pi * n) / 2 + 1).toInt
        println(r)
    }
}

File: JavaScript/3096.js
var input = require('fs').readFileSync('/dev/stdin', 'utf8');
var lines = input.split('\n');

var n = parseInt(lines.shift());
   
console.log(Math.floor(n*Math.log10(n/Math.E) + Math.log10(2*Math.PI*n)/2.0) + 1);